### PR TITLE
Add integration tests with cypress for frontend

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
Add integration tests, dashboard needs more tests after filters and histogram are implemented

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-notifications/.gitignore [hunk 1]
```diff
@@ -6,6 +6,6 @@ node_modules
 yarn-error.log
 /coverage/
 .DS_Store
-/cypress/screenshots/
-/cypress/videos/
-target
\ No newline at end of file
+/.cypress/screenshots/
+/.cypress/videos/
+target
```
Comment (zhongnansu) [inline dashboards-notifications/.gitignore:11]: do we still have target, I thought after 7.10, kibana removes it

File: dashboards-notifications/.cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// Cypress.Commands.add("dismiss", { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
+
+Cypress.Commands.overwrite('visit', (originalFn, url, options) => {
+  // Add the basic auth header when security enabled in the OpenSearch cluster
+  // https://github.com/cypress-io/cypress/issues/1288
+  if (Cypress.env('security_enabled')) {
```
Comment (zhongnansu) [inline dashboards-notifications/.cypress/support/commands.js:43]: just curious, does this mean we add cypress test for cluster with security enabled?

**Review by zhongnansu — APPROVED** [review]
lgtm

File: dashboards-notifications/.cypress/support/commands.js (hunk 2 above)
Comment (davidcui-amzn) [reply dashboards-notifications/.cypress/support/commands.js:43]: Yes, same as in Reporting here https://github.com/opensearch-project/dashboards-reports/blob/72705e2dfcad760c5de7609891700aa11d767884/dashboards-reports/.cypress/support/commands.js#L58

**Review by davidcui-amzn — APPROVED** [review]
LGTM

File: dashboards-notifications/.cypress/support/commands.js (hunk 2 above)
Comment (zhongnansu) [reply dashboards-notifications/.cypress/support/commands.js:43]: @davidcui-amzn @joshuali925 what's the command to run cypress test with secuirty enabled cluster? Also, is there a way to add cypress test in github CI?

**Pull request merged**